### PR TITLE
Add a missing code to skip NSG deletion if the vnet is unmanaged

### LINKED ISSUE #1154: CAPZ is trying to delete NSG when deleting unmanaged cluster
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

When try to delete an unmanaged cluster, CAPZ is trying to delete the NSG, which then failed because subnet is still there. Should skip NSG deletion for unmanaged clusters. 

```
[capz-controller-manager-5f6b77c4bf-n4gm7] 1 controller.go:257] controller-runtime/controller "msg"="Reconciler error" "error"="error deleting AzureCluster cluster-6017c0fd9dc6871a9295b8c5/ztkjx: failed to delete network security group: failed to delete security group ztkjx-static-controlplane-nsg in resource group automation-teams: network.SecurityGroupsClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code=\"InUseNetworkSecurityGroupCannotBeDeleted\" Message=\"Network security group /subscriptions/8710ff2b-e468-434a-9a84-e522999f6b81/resourceGroups/automation-teams/providers/Microsoft.Network/networkSecurityGroups/ztkjx-static-controlplane-nsg cannot be deleted because it is in use by the following resources: /subscriptions/8710ff2b-e468-434a-9a84-e522999f6b81/resourceGroups/automation-teams/providers/Microsoft.Network/virtualNetworks/ztkjx-static-vnet/subnets/ztkjx-static-controlplane-subnet. In order to delete the Network security group, remove the association with the resource(s). To learn how to do this, see aka.ms/deletensg.\" Details=[]" "controller"="azurecluster" "name"="ztkjx" "namespace"="cluster-6017c0fd9dc6871a9295b8c5"
```

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: v0.4.11
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Evalle**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adds a missing code to skip NSG deletion if the vnet is unmanaged

**Which issue(s) this PR fixes**
Fixes #1154

**Special notes for your reviewer**:
both `make lint` and `make test` passed locally

**Release note**:
```release-note
Skip NSG deletion if the vnet is unmanaged
```

### CODE REVIEW

File: cloud/services/securitygroups/securitygroups.go [hunk 1]
```diff
@@ -140,6 +140,11 @@ func (s *Service) Delete(ctx context.Context) error {
 	ctx, span := tele.Tracer().Start(ctx, "securitygroups.Service.Delete")
 	defer span.End()
 
+	if !s.Scope.IsVnetManaged() {
+		s.Scope.V(4).Info("Skipping network security group reconcile in custom VNet mode")
```
Comment (CecileRobertMichon) [inline cloud/services/securitygroups/securitygroups.go:144]:
```suggestion
		s.Scope.V(4).Info("Skipping network security group delete in custom VNet mode")
```

**Comment by CecileRobertMichon** [comment]
@Evalle please add a release note in description, this is an important fix and definitely one we want to call out in release notes

File: cloud/services/securitygroups/securitygroups.go [hunk 2]
```diff
@@ -140,6 +140,11 @@ func (s *Service) Delete(ctx context.Context) error {
 	ctx, span := tele.Tracer().Start(ctx, "securitygroups.Service.Delete")
 	defer span.End()
 
+	if !s.Scope.IsVnetManaged() {
```
Comment (CecileRobertMichon) [inline cloud/services/securitygroups/securitygroups.go:143]: let's add a unit test case for this as well

Comment (CecileRobertMichon) [reply cloud/services/securitygroups/securitygroups.go:143]: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/cloud/services/securitygroups/securitygroups_test.go#L255

Comment (Evalle) [reply cloud/services/securitygroups/securitygroups.go:143]: I've added a unit-test (basically copy-pasted the one from `reconcile`) let me know what you think. Thanks.

**Review by CecileRobertMichon — APPROVED** [review]
lgtm

Please squash the commits

**Comment by Evalle** [comment]
Done, thank for your review @CecileRobertMichon !

File: azure/services/securitygroups/securitygroups_test.go [hunk 3]
```diff
@@ -307,6 +307,13 @@ func TestDeleteSecurityGroups(t *testing.T) {
 				m.Delete(gomockinternal.AContext(), "my-rg", "nsg-two")
 			},
 		},
+		{
+			name: "skipping network security group delete in custom VNet mode",
+			expect: func(s *mock_securitygroups.MockNSGScopeMockRecorder, m *mock_securitygroups.MockclientMockRecorder) {
+				s.IsVnetManaged().Return(false)
```
Comment (CecileRobertMichon) [inline azure/services/securitygroups/securitygroups_test.go:313]: you need to add this expect call to the other test cases with a return value of true

`s.IsVnetManaged().Return(true)`

Comment (CecileRobertMichon) [reply azure/services/securitygroups/securitygroups_test.go:313]: line 285 and 304

Comment (Evalle) [reply azure/services/securitygroups/securitygroups_test.go:315]: Added, thanks

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**